### PR TITLE
Add Nuvoton npcx7 series GPIO pin list

### PR DESCRIPTION
**Pull request by vphirema**
Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

### CODE REVIEW

File: soc/arm/nuvoton_npcx/npcx7/soc.h [hunk 1]
```diff
@@ -20,4 +20,157 @@
...
+#define NPCX_GPIO_A6	(6U)
+#define NPCX_GPIO_A7	(7U)
+
+/* GPIO_B */
+#define NPCX_GPIO_B0	(0U)
+#define NPCX_GPIO_B1	(1U)
+#define NPCX_GPIO_B2	(2U)
+#define NPCX_GPIO_B3	(3U)
+#define NPCX_GPIO_B4	(4U)
+#define NPCX_GPIO_B5	(5U)
```
Comment (MulinChao) [inline soc/arm/nuvoton_npcx/npcx7/soc.h:136]: Please don't define GPIO85. It is a special pin for PSL Output and driven high by default.

Comment (vphirema) [reply soc/arm/nuvoton_npcx/npcx7/soc.h:136]: sure , it is not in the list

**Comment by MulinChao** [comment]
That's my personal opinion. The reason I don't put these definitions in npcx7 GPIO PR is that I'd like to let the node in the device tree file of board folded to handle it.  For example,  here is a simple power sequence application.

**In device tree file:**
```
	pwr_seq_keys {
		compatible = "gpio-keys";
		gpio_d7: gpiod7 {
			label = "POWER_PCH_EC_SLP_SUS_N";
			gpios = <&gpiod 7 GPIO_ACTIVE_LOW>;
		};
	};

	aliases {
		power_pch_ec_slp_sus_n = &gpio_d7;
		power_vr_ec_rsmrst_pwrgd = &gpio_e2;
		...
	};
```
	
**In power seqence routine:**
```
#define NODE_PCH_EC_SLP_SUS_N DT_ALIAS(power_pch_ec_slp_sus_n)

#define PWR_GPIO_LABEL(node)  DT_GPIO_LABEL(node, gpios)
#define PWR_GPIO_PIN(node)	  DT_GPIO_PIN(node, gpios)
#define PWR_GPIO_FLAGS(node)  DT_GPIO_FLAGS(node, gpios)

const struct device *pin_slp_sus_n = device_get_binding(PWR_GPIO_LABEL(NODE_PCH_EC_SLP_SUS_N));
int ret = gpio_pin_configure(pin_slp_sus_n, PWR_GPIO_PIN(NODE_PCH_EC_SLP_SUS_N), PWR_GPIO_FLAGS(NODE_PCH_EC_SLP_SUS_N));
```

The benefit is we needn't change the c source for different boards if the PCH is the same.

**Comment by jettr** [comment]
> The benefit is we needn't change the c source for different boards if the PCH is the same

I do like the example you provided. Hopefully we can standardize the "power sequencing IO" device in Zephyr upstream. I think that would be idea.

**Comment by vphirema** [comment]
ACK.
I think we still need this change as any GPIOs can be used for any purpose.
May be we can add a macro which will convert the pin to binding string & pin

**Comment by vphirema** [comment]
Hi Mulin,
I am not familiar with DTS file. As Aliases dont support underscore, is it possible to do 
device_get_binding(SOME_MACRO(PCH_EC_SLP_SUS_L))

        pow_seq_gpio_keys {
                compatible = "gpio-keys";

                /* Interrupts */
                PCH_EC_SLP_SUS_L: GPIOD7 {
                        label = "SLP_SUS_L";
                        gpios = <&gpiod 7 GPIO_ACTIVE_LOW>;
                };

**Comment by MulinChao** [comment]
> Hi Mulin,
> I am not familiar with DTS file. As Aliases dont support underscore, is it possible to do
> device_get_binding(SOME_MACRO(PCH_EC_SLP_SUS_L))
> 
> ```
>     pow_seq_gpio_keys {
>             compatible = "gpio-keys";
> 
>             /* Interrupts */
>             PCH_EC_SLP_SUS_L: GPIOD7 {
>                     label = "SLP_SUS_L";
>                     gpios = <&gpiod 7 GPIO_ACTIVE_LOW>;
>             };
> ```

Hi Vijay:
If you want to get label, pin, and flag without alias, I think DT_NODELABEL(PCH_EC_SLP_SUS_L) might help.
```
pow_seq_gpio_keys {
		compatible = "gpio-keys";

		/* Interrupts */
		pch_ec_sus_slp: gpiod7 {
			label = "SLP_SUS_L";
			gpios = <&gpiod 7 GPIO_ACTIVE_LOW>;
		};
};
```

`printk("%s", DT_GPIO_LABEL(DT_NODELABEL(pch_ec_sus_slp), gpios));`

**Comment by vphirema** [comment]
Thank you Mulin,
Blow command worked, also caps & underscore dont work
.name = DT_GPIO_LABEL(DT_NODELABEL(node), gpios)

**Comment by vphirema** [comment]
this change is not needed as suggested by Mulin hence closing it

**Pull request closed without merging**